### PR TITLE
CBL-442 (partial): enable standard error messages

### PR DESCRIPTION
**Pull request by bmeike**
The standard error messages are now available.  Just need to find them and use them.

### CODE REVIEW

File: src/shared/main/java/com/couchbase/lite/internal/support/Log.java [hunk 1]
```diff
@@ -408,6 +332,32 @@ public static void setC4LogLevel(EnumSet<LogDomain> domains, LogLevel level) {
         }
     }
 
+    private static void log(
+        @NonNull LogLevel level,
+        @NonNull LogDomain domain,
+        @Nullable Throwable err,
+        @Nullable String msg,
+        Object... args) {
+        String message = lookupErrorMessage(msg);
```
Comment (pasin) [inline src/shared/main/java/com/couchbase/lite/internal/support/Log.java]: As msg is Nullable, it seems like the message could be null. It's hard to know from this PR what is the result of null message. @bmeike can you review if we need to handle null message?

File: src/shared/main/java/com/couchbase/lite/internal/support/Log.java [hunk 2]
```diff
@@ -408,6 +332,32 @@ public static void setC4LogLevel(EnumSet<LogDomain> domains, LogLevel level) {
...
+    }
+
+    private static String lookupErrorMessage(String error) {
+        final String message = errorMessages.get(error);
+        return (message == null) ? error : message;
+    }
+
+    private static String formatMessage(String msg, Object... args) {
+        try { return String.format(msg, (Object[]) args); }
+        catch (IllegalFormatException | FormatterClosedException ignore) { }
```
Comment (jayahariv) [inline src/shared/main/java/com/couchbase/lite/internal/support/Log.java]: not seeing the Locale.ENGLISH setup, also exceptions are not handled. is this intentional?

Comment (bmeike) [reply src/shared/main/java/com/couchbase/lite/internal/support/Log.java]: Good call on the Local.  Will fix.  Yes.  If there is a formatting exception, the code will simply return the value of `msg` without trying to fill in the template.  I'm thinking that's the best I can do...

File: src/shared/test/java/com/couchbase/lite/LogTest.java [hunk 3]
```diff
@@ -298,6 +310,38 @@ public void testFileLoggingHeader() throws Exception {
...
+        Database.log.setCustom(logger);
+
+        Log.d(LogDomain.DATABASE, "TEST DEBUG");
+        assertEquals("TEST DEBUG", logger.getMessage());
+
+        Log.d(LogDomain.DATABASE, "TEST DEBUG", new Exception("whoops"));
+        assertEquals("TEST DEBUG (java.lang.Exception: whoops)", logger.getMessage());
+
+        Log.d(LogDomain.DATABASE, "TEST DEBUG %s %d %f", "arg", 1, 3.0F);
+        assertEquals("TEST DEBUG arg 1 3.000000", logger.getMessage());
```
Comment (jayahariv) [inline src/shared/test/java/com/couchbase/lite/LogTest.java]: is it always sure to have exactly 6 decimal places in all places?

File: src/shared/main/java/com/couchbase/lite/internal/support/Log.java (hunk 1 above)
Comment (bmeike) [reply src/shared/main/java/com/couchbase/lite/internal/support/Log.java]: Nice catch.  There used to be a weird version of `w(domain, err)` that did not take a message.  Nobody used it, so I deleted it.  Message is now `@NonNull`

File: src/shared/test/java/com/couchbase/lite/LogTest.java (hunk 3 above)
Comment (bmeike) [reply src/shared/test/java/com/couchbase/lite/LogTest.java]: Yeah.... I can fix that with format `%.2f`.  Will do.

**Pull request merged**